test/UserService.Business.UnitTests/GetUserCommandTests.cs

```diff
@@ -0,0 +1,284 @@
+                _mocker.GetMock<IUserRepository>().Object,
+                _mocker.GetMock<IUserResponseMapper>().Object,
+                _mocker.GetMock<IRequestClient<IGetDepartmentRequest>>().Object,
+                _mocker.GetMock<IRequestClient<IGetPositionRequest>>().Object,
+                _mocker.GetMock<IRequestClient<IGetUserProjectsInfoRequest>>().Object,
+                _mocker.GetMock<IRequestClient<IGetFileRequest>>().Object);
+        }
+
+        [Test]
+        public void ThrowExseptionWhenFilterIsNull()
+        {
+            _filter = null;
+
+            Assert.Throws<BadRequestException>(() => _command.Execute(_filter));
+            _mocker.Verify<IUserRepository, DbUser>(x => x.Get(It.IsAny<GetUserFilter>()), Times.Never());
+            _mocker.Verify<IUserResponseMapper, UserResponse>(x => x.Map(
+                It.IsAny<DbUser>(),
+                It.IsAny<DepartmentInfo>(),
+                It.IsAny<PositionInfo>(),
+                It.IsAny<List<ProjectInfo>>(),
+                It.IsAny<List<ImageInfo>>(),
+                It.IsAny<GetUserFilter>(),
+                It.IsAny<List<string>>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void ThrowExseptionWhenFilterDoNotContainIdAndNameAndEmailData()
+        {
+            _filter.UserId = null;
+            _filter.Name = null;
+            _filter.Email = null;
+
+            Assert.Throws<BadRequestException>(() => _command.Execute(_filter));
+            _mocker.Verify<IUserRepository, DbUser>(x => x.Get(It.IsAny<GetUserFilter>()), Times.Never());
+            _mocker.Verify<IUserResponseMapper, UserResponse>(x => x.Map(
+                It.IsAny<DbUser>(),
+                It.IsAny<DepartmentInfo>(),
+                It.IsAny<PositionInfo>(),
+                It.IsAny<List<ProjectInfo>>(),
+                It.IsAny<List<ImageInfo>>(),
+                It.IsAny<GetUserFilter>(),
+                It.IsAny<List<string>>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void ThrowNotFoundExceptionWhenUserIsNotFound()
+        {
+            DbUser emptyDbUser = null;
+            _mocker
+                .Setup<IUserRepository, DbUser>(x => x.Get(It.IsAny<GetUserFilter>()))
+                .Returns(emptyDbUser);
+
+            Assert.Throws<NotFoundException>(() => _command.Execute(_filter));
+            _mocker.Verify<IUserRepository, DbUser>(x => x.Get(It.IsAny<GetUserFilter>()), Times.Once());
+            _mocker.Verify<IUserResponseMapper, UserResponse>(x => x.Map(
+                It.IsAny<DbUser>(),
+                It.IsAny<DepartmentInfo>(),
+                It.IsAny<PositionInfo>(),
+                It.IsAny<List<ProjectInfo>>(),
+                It.IsAny<List<ImageInfo>>(),
+                It.IsAny<GetUserFilter>(),
+                It.IsAny<List<string>>()),
+                Times.Never);
+        }
+
+        [Test]
+        public void ReturnResponse()
```

Comment:
rename to ShouldSuccessfulReturnResponse or smth like that

Reply:
Add verify for request client

Reply:
Done